gprofiler/spark/spark_collector.py

```diff
@@ -806,15 +754,18 @@ def _collect_loop(self) -> None:
                         break
 
             if self._spark_sampler is not None:
-                metrics = list(self._spark_sampler.collect())
-                timestamp = datetime.now(tz=timezone.utc)
+                collected = self._spark_sampler.collect()
+                # No need to submit samples that don't actually have a value:
+                samples = tuple(filter(lambda s: s.value is not None, collected))
+                snapshot = MetricsSnapshot(datetime.now(tz=timezone.utc), samples)
+                # _client and _storage_dir are mutually exclusive, so we will never bake the payload twice
```

Comment:
Are they? You can pass both.

Reply:
`_client` => `--upload-results`, and `_storage_dir` => `--output-dir`, and the two flags are mutually exclusive, so de facto so are the fields. No?

Reply:
They are not mutual exclusive o.O